android/src/main/java/com/amplitude/android/Configuration.kt

```diff
@@ -39,8 +42,10 @@ open class Configuration @JvmOverloads constructor(
     var minTimeBetweenSessionsMillis: Long = MIN_TIME_BETWEEN_SESSIONS_MILLIS,
     var trackingSessionEvents: Boolean = true,
     override var identifyBatchIntervalMillis: Long = IDENTIFY_BATCH_INTERVAL_MILLIS,
-    override var identifyInterceptStorageProvider: StorageProvider = AndroidStorageProvider()
-) : Configuration(apiKey, flushQueueSize, flushIntervalMillis, instanceName, optOut, storageProvider, loggerProvider, minIdLength, partnerId, callback, flushMaxRetries, useBatch, serverZone, serverUrl, plan, ingestionMetadata, identifyBatchIntervalMillis, identifyInterceptStorageProvider) {
+    override var identifyInterceptStorageProvider: StorageProvider = AndroidStorageProvider(),
+    override var identityStorageProvider: IdentityStorageProvider = FileIdentityStorageProvider(),
+    initializers: List<Initializer>? = null
```

Comment:
> Thanks @falconandy. I see what you mean about the complexity of the ID logic. I'm trying to think if there is anything else we can do to prevent adding the new concept of "initializers", but nothing is coming to mind.

Given the complexities with id, and now that Migration code/plugin is a part of the SDK codebase, I think instead of adding the new concept of `initializers` we can instead include the migration logic by default and toggle it on/off via Configuration. In this case the migration logic no longer needs to use the Plugin interface.

Ideally if `migrateLegacyData` is `false` (the default) no SQLLite code is run at all.

```
val amplitude = Amplitude(Configuration(migrateLegacyData: true));
```

@falconandy can you please refactor accordingly. Thank you 🙏

Reply:
Refactored, added the `migrateLegacyData` flag (default is `false`)